iter/filter.go

```diff
@@ -36,3 +36,28 @@ func Exclude[T any](iter Iterator[T], fun func(T) bool) *FilterIter[T] {
 	inverse := func(t T) bool { return !fun(t) }
 	return &FilterIter[T]{iter, inverse}
 }
+
+type FilterMapIter[T any, U any] struct {
```

Comment:
Please include a compile time guarantee that the filter map iterator implements the iterator interface, see line 31.